sdk/aqueduct/integrations/sql_integration.py

```diff
@@ -150,32 +152,59 @@ def sql(
             extract_params = RelationalDBExtractParams(
                 query=extract_params,
             )
+        elif isinstance(extract_params, list):
+            for q in extract_params:
+                assert isinstance(
+                    q, str
+                ), "When using a list of queries, it must be a list of strings."
+
+            if len(extract_params) == 1:
+                extract_params = RelationalDBExtractParams(
+                    query=extract_params[0],
+                )
+            else:
+                extract_params = RelationalDBExtractParams(
+                    queries=extract_params,
+                )
+        elif isinstance(
+            extract_params, RelationalDBExtractParams
+        ):  # query is a RelationalDBExtractParams object
+            if int(bool(extract_params.query)) + int(bool(extract_params.queries)) != 1:
+                raise Exception(
+                    "For a RelationalDBExtractParams object, exactly one of .query or .queries fields should be set."
+                )
 
         # Find any tags that need to be expanded in the query, and add the parameters that correspond
         # to these tags as inputs to this operator. The orchestration engine will perform the replacement at runtime.
         sql_input_artifact_ids = []
+        matches = []
         if extract_params.query is not None:
             matches = re.findall(TAG_PATTERN, extract_params.query)
-            for match in matches:
-                param_name = match.strip(" {}")
-                param_artifact = self._dag.get_artifact_by_name(param_name)
-                if param_artifact is None:
-                    # If it is a built-in tag, we can ignore it for now, since the python operators will perform the expansion.
-                    if param_name in BUILT_IN_EXPANSIONS:
-                        continue
-
-                    raise InvalidUserArgumentException(
-                        "There is no parameter defined with name `%s`." % param_name,
-                    )
-
-                # Check that the parameter corresponds to a string value.
-                if param_artifact.type != ArtifactType.STRING:
-                    raise InvalidUserArgumentException(
-                        "The parameter `%s` must be defined as a string. Instead, got type %s"
-                        % (param_name, param_artifact.type)
-                    )
-
-                sql_input_artifact_ids.append(param_artifact.id)
+
+        if extract_params.queries is not None:
+            for q in extract_params.queries:
+                matches.extend(re.findall(TAG_PATTERN, q))
```

Comment:
To simplify this, can we make a separate list variable containing all the queries, and just have one loop over that? In the single element case, this would just be a list of size 1.